modules/elasticsearch/src/test/java/com/github/jeanbaptistewatenberg/junit5kubernetes/elasticsearch/ElasticSearchPodTest.java

```diff
@@ -1,19 +1,101 @@
 
 public class ElasticSearchPodTest {
+    private static final String ELASTICSEARCH_VERSION = Version.CURRENT.toString();
+    public static final String ELASTICSEARCH_IMAGE_NAME = "docker.elastic.co/elasticsearch/elasticsearch";
+    private static final String ELASTICSEARCH_IMAGE = ELASTICSEARCH_IMAGE_NAME+":"+ELASTICSEARCH_VERSION;
 
     @Test
     void shouldCreateElasticSearchPod() {
         try(ElasticSearchPod pod = new ElasticSearchPod()){
+            //S
             pod.create();
-            assertThat(pod.getHttpUrl()).isEqualTo(
+            //V
+            assertThat(pod.getHttpHostAddress()).isEqualTo(
                     String.format("http://%s:%d", pod.getObjectHostIp(), pod.getHttpPort()));
+        }
+    }
 
+    @Test
+    void shouldCreateElasticSearchDefaultImageXpackShouldBeAvailable() {
+        try(ElasticSearchPod pod = new ElasticSearchPod()){
+            //S
+            pod.create();
+            final RestHighLevelClient restHighLevelClient = new RestHighLevelClient(RestClient.builder(
+                    new HttpHost(pod.getObjectHostIp(), pod.getHttpPort(), "http")
+            ));
+            //E
+            final XPackInfoResponse xpack = restHighLevelClient.xpack().info(new XPackInfoRequest(), RequestOptions.DEFAULT);
+            //V
+            assertThat(xpack).isNotNull();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
 
+    @Test
+    void shouldCreateElasticSearchVersionXpackShouldBeAvailable() {
+        try(ElasticSearchPod pod = new ElasticSearchPod(ELASTICSEARCH_IMAGE)){
+            //S
+            pod.create();
+            final RestHighLevelClient restHighLevelClient = new RestHighLevelClient(RestClient.builder(
+                    new HttpHost(pod.getObjectHostIp(), pod.getHttpPort(), "http")
+            ));
+            //E
+            final XPackInfoResponse xpack = restHighLevelClient.xpack().info(new XPackInfoRequest(), RequestOptions.DEFAULT);
+            //V
+            assertThat(xpack).isNotNull();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    void shouldCreateElasticSearchOssImageXpackShouldNotBeAvailable() {
+        try(ElasticSearchPod pod = new ElasticSearchPod("docker.elastic.co/elasticsearch/elasticsearch-oss"+":"+ELASTICSEARCH_VERSION)){
+            //S
+            pod.create();
+            final RestHighLevelClient restHighLevelClient = new RestHighLevelClient(RestClient.builder(
+                    new HttpHost(pod.getObjectHostIp(), pod.getHttpPort(), "http")
+            ));
+            //E
+            final ElasticsearchStatusException elasticsearchStatusException = assertThrows(ElasticsearchStatusException.class, () -> restHighLevelClient.xpack().info(new XPackInfoRequest(), RequestOptions.DEFAULT));
+            //V
+            assertThat(elasticsearchStatusException.status()).isEqualTo(RestStatus.BAD_REQUEST);
+        }
+    }
+
```

Comment:
Im not sure about the relevance of this test which is testing an implementation detail of open source version of elastic docker image and not the actual code of this module, except if you want to test usage of specific docker image, but then it is duplicated with the previous test that could test if the cluster version is exactly the expected one

```suggestion

```